From 688747ce25ea44e67255334a067a6874a7c59ce4 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Fri, 27 Aug 2021 16:20:06 +0530
Subject: [PATCH 6/9] octeontx2-vf: Fix SQB threshold check while submitting
 SQE

commit e08774b90bbc55d7b63dae7b8d41ab29931145a7 from
git@git.assembla.com:cavium/WindRiver.linux.git

In VF netdev driver, fix SQB threshold checks before submitting
SQE to HW. These changes are already upstreamed but missing in
internal driver.

Change-Id: Ic0b7029a80f91ce82125a3aafe575ce3105d74d3
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/60036
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
index 8d4a04de81dd..2c2013ee3a66 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
@@ -395,12 +395,9 @@ static netdev_tx_t otx2vf_xmit(struct sk_buff *skb, struct net_device *netdev)
 
 		/* Check again, incase SQBs got freed up */
 		smp_mb();
-		if ((sq->num_sqbs - *sq->aura_fc_addr) > 1)
-			netif_tx_start_queue(txq);
-		else
-			netdev_warn(netdev,
-				    "%s: No free SQE/SQB, stopping SQ%d\n",
-				     netdev->name, qidx);
+		if (((sq->num_sqbs - *sq->aura_fc_addr) * sq->sqe_per_sqb)
+							> sq->sqe_thresh)
+			netif_tx_wake_queue(txq);
 
 		return NETDEV_TX_BUSY;
 	}
-- 
2.31.1

